src/ghci/stdout.rs

```diff
@@ -43,36 +45,28 @@ impl GhciStdout {
         ]);
         let data = self
             .reader
-            .read_until(&bootup_patterns, WriteBehavior::Write, &mut self.buffer)
+            .read_until(&mut ReadOpts {
+                end_marker: &bootup_patterns,
+                find: FindAt::LineStart,
+                writing: WriteBehavior::Write,
+                buffer: &mut self.buffer,
+            })
             .await?;
         tracing::debug!(data, "ghci started, saw version marker");
 
-        // We know that we'll get _one_ `ghci> ` prompt on startup.
-        let init_prompt_patterns = AhoCorasick::from_anchored_patterns(["ghci> "]);
-        let messages = self.prompt(Some(&init_prompt_patterns)).await?;
-        tracing::debug!("Saw initial `ghci> ` prompt");
```

Comment:
This is unreliable, so I extended the prompt matching logic to allow matching a prompt in the middle of the line instead. Then, we unconditionally send our `:set prompt` command when we start `ghci`, so we see something like `ghci> ###~GHCID-NG-PROMPT~###` on stdout and match that.